src/control/server/config.go

```diff
@@ -394,6 +394,15 @@ func saveActiveConfig(log logging.Logger, config *Configuration) {
 
 // Validate asserts that config meets minimum requirements.
 func (c *Configuration) Validate(log logging.Logger) (err error) {
+	// config without servers is valid when initially discovering hardware
+	// prior to providing a fully populated configu
+	if len(c.Servers) == 0 {
+		log.Infof("No %ss in configuration, %s starting in discovery mode",
+			DataPlaneName, ControlPlaneName)
+		c.Servers = nil
+		return nil
+	}
```

Comment:
Is this enough to allow the server to start up in discovery mode? I believe that there is another check for the fabric provider which will fail when it's blank.

Reply:
it's working so I think it sufficient

Reply:
Oh, right. I was thinking the fabric validation happened elsewhere, but it's right below this section.